router/router.go

```diff
@@ -0,0 +1,160 @@
+package router
+
+import (
+	"errors"
+	"strings"
+)
+
+type Route struct {
+	// children route can be nil
+	children map[string]*Route
+	// variableRoute is prepare for route like `:name`
+	variableRoute *Route
+	// ownHandler means this Route has route, so not found handler would be 403(wrong method),
+	// else is 404
+	ownHandler bool
+	// wildcardMethodHandlers is the handler of route `*path`
+	wildcardMethodHandlers map[string]Handler
+	// methodHandlers stores map Method to handler of this route
+	methodHandlers map[string]Handler
+	// optionsHandlerBuilder stores a special handler for OPTION method handling
+	optionsHandlerBuilder *optionsHandlerBuilder
+	optionHandler         OptionsHandler
+	notAllowHandler       func(method string) Handler
+}
+
+// OptionsHandler interface define an interface that helps router automatically create an OPTION method handler for you
+type OptionsHandler interface {
+	Build(allowMethods string) Handler
+}
+
+type Handler interface {
+	WildcardIndex(int) error // maybe we should fall failed while we emit a wildcard index onto a handler don't do it?
```

Comment:
sorry, I cannot understand this comment 🙈  can you explain more?

Reply:
Ok, I got what the problem is, originally I thought some handler might not want to support Wildcard route if true then they should return an error at here, but seems very wrong concern. Those handlers do not want to support wildcard can just simply empty the implementation or just panic.

Anyway, I would remove that error from `WildcardIndex`